ete3/tools/ete_evol.py

```diff
@@ -439,6 +447,15 @@ def remove_duplicated_marks(nodes, marks, tree):
         del(marks[bad])
         del(nodes[bad])
 
+
+def name_model(tree, base_name):
+    """
+    transform the name string into summary of its name and a digestion of the
+    full name
+    """
+    return base_name[:12] + '__' + md5(tree.write() + base_name).hexdigest()
```

Comment:
this might lead to different md5 even with the same tree. Use tree.et_topology_id(attr="name") instead?